src/charm.py

```diff
@@ -193,12 +221,26 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
         self._handle_status_update_config(event)
 
     def _on_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
-        if self.unit.is_leader():
-            logger.info("This app's public ingress URL: %s", event.url)
+        self._handle_status_update_config(event)
 
     def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent) -> None:
-        if self.unit.is_leader():
-            logger.info("This app no longer has ingress")
+        self._handle_status_update_config(event)
+
+    def _on_oauth_info_changed(self, event: OAuthInfoChangedEvent) -> None:
+        self._handle_status_update_config(event)
+
+    def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
+        self._container.push(
+            CA_CERT_DIR_PATH / f"{event.relation_id}.pem", event.ca, make_dirs=True
+        )
+        self._handle_status_update_config(event)
+
+    def _on_certificate_removed(self, event: CertificateRemovedEvent) -> None:
+        try:
+            self._container.remove_path(CA_CERT_DIR_PATH / f"{event.relation_id}.pem")
+        except PathError:
+            pass
+        self._handle_status_update_config(event)
```

Comment:
one thing I am not sure of is whether we should add the logic for pushing the cert to the container in `_handle_status_update_config` so that it always runs (holistic approach). It would make the logic a little more complicated, but it would also make it less error prone